splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/sequence/AbstractSequence.java

```diff
@@ -23,14 +23,16 @@
 import java.io.ObjectOutput;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public abstract class AbstractSequence implements Sequence, Externalizable{
     protected final AtomicLong remaining=new AtomicLong(0l);
     protected final AtomicLong currPosition=new AtomicLong(0l);
```

Comment:
Please consider making `remaining` and `currPosition` non-atomic if R/W accesses to these variables are protected with `rwLock`, I haven't checked all of their usages.